middleware.js

```diff
@@ -92,6 +93,25 @@ function errorHandler (err, req, res, next) {
 }
 
 
+async function _getRequestSessionUser(req, res, next) {
+  if (req.session && req.session.user) {
+    // make sure the user object has all subscribers and email_addresses properties
+    return DB.getSubscriberById(req.session.user.id);
+  }
+  return null;
+}
+
+async function requireSessionUser(req, res, next) {
+  const user = await _getRequestSessionUser(req);
+  if (!user) {
+    return res.redirect("/oauth/init");
```

Comment:
Just ran this locally and I was redirected, but when I added UTM parameters, they weren't passed along to FxA. I think we just need to check for any search params and append them to the /ouath/init route.

Reply:
Updated to preserve utm params.